test/blocks/caas/utils.test.js

```diff
@@ -779,6 +779,88 @@ describe('getCountryAndLang', () => {
     });
   });
 
+  it('should use GEO IP for langFirst when not news source (regardless of URL country)', async () => {
+    // Set metadata to enable langFirst
+    const metaLangFirst = document.createElement('meta');
+    metaLangFirst.setAttribute('name', 'langfirst');
+    metaLangFirst.setAttribute('content', 'true');
+    document.head.appendChild(metaLangFirst);
+
+    setConfig({
+      pathname: '/en/blah.html',
+      locales: { '': { ietf: 'en-US' } },
+    });
+
+    const expected = await getCountryAndLang({
+      autoCountryLang: true,
+      source: ['hawks'],
+    });
+
+    // Should ALWAYS try GEO IP first (not just when country is 'xx')
+    // Will get GEO IP country or fall back to URL path country
+    expect(expected.country).to.not.eq('xx');
+    expect(expected.language).to.eq('en');
+
+    document.head.removeChild(metaLangFirst);
+  });
+
+  it('should NOT use GEO IP for news source even with langFirst', async () => {
+    // Set metadata to enable langFirst
+    const metaLangFirst = document.createElement('meta');
+    metaLangFirst.setAttribute('name', 'langfirst');
+    metaLangFirst.setAttribute('content', 'true');
+    document.head.appendChild(metaLangFirst);
```

Comment:
this block is repeated for each `it`, seems like a good candidate for `beforeEach`, perhaps grouping this in their own `describe`